nosnoc/problem.py

```diff
@@ -356,6 +352,21 @@ def forward_simulation(self, ocp: NosnocOcp, Uk: ca.SX) -> None:
             X_fe.append(self.w[self.ind_x[-1]])
         elif opts.irk_representation == IrkRepresentation.DIFFERENTIAL_LIFT_X:
             X_fe = [self.w[ind] for ind in self.ind_x]
+
+        return X_fe
+
+    def forward_simulation(self, ocp: NosnocOcp, Uk: ca.SX) -> None:
+        opts = self.opts
+        model = self.model
+
+        # setup X_fe: list of x values on fe, initialize X_end
+        # TODO: clean up
```

Comment:
what do you mean here?

Reply:
There is probably a cleaner way to set up the following conditionals. I can remove this comment probably, because this is already a slightly cleaner version than what I started with.